crates/rust/src/interface.rs

```diff
@@ -1004,34 +1004,25 @@ unsafe fn call_import(params: *mut u8, results: *mut u8) -> u32 {{
             needs_cleanup_list,
             src,
             handle_decls,
-            async_result_name,
             ..
         } = f;
         if async_ {
+            let async_support = self.r#gen.async_support_path();
+            uwriteln!(self.src, "{async_support}::start_task(async move {{");
             if needs_cleanup_list {
-                let vec = self.path_to_vec();
-                uwriteln!(self.src, "let mut cleanup_list = {vec}::new();");
+                uwriteln!(self.src, "let mut cleanup_list = Vec::new();");
```

Comment:
Why do we not need to use `self.path_to_vec()` here?

Reply:
Oh this is my mistake, we do still need that and this just got lost in some refactorings I made along the way and didn't double-check the diff at the end.